### PR TITLE
Configuration for test-kitchen

### PR DESCRIPTION
**Pull request by geewiz**
This configuration file defines defaults for test-kitchen projects:
- Use Docker driver
- Pre-install Chef 12 and Busser packages for faster spin-up

@freistil/ops This configuration is highly recommended!

### CODE REVIEW

File: kitchen/config.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+driver_plugin: docker
+driver_config:
+  provision_command:
+    - curl -L https://www.opscode.com/chef/install.sh | bash -s -- -v 12.0.3
```
Comment (muhh) [inline kitchen/config.yml]: This installs an outdated chef version. And it pins the version. I'd prefer to just install the latest as default.

**Comment by geewiz** [comment]
Pinning the version ensures that Docker only has to install it once and then use its cache for every new container. In think that in this context, test performance is more important than running the bleeding edge (of which I don't see the advantage).

**Comment by muhh** [comment]
No need for bleeding edge. But it should at least be the current version we have rolled out everywhere.
No need to for keeping using 16 month old buggy chef-clients in development.

**Comment by geewiz** [comment]
Ok. I'm testing with 12.9.41 and will push an update as soon as it's finished.

**Comment by geewiz** [comment]
At least with the latest version, Busser (integration tests) throws an exception. That's why I added the version constraint (that I found on the web) in the first place; when that worked, I left it at the old version. I'll try and find out up to which Chef version Busser doesn't break.

**Comment by geewiz** [comment]
It works with 12.6.0, the version we use in production. So I think it makes sense using that by default.

**Pull request merged**